cli/src/lib.rs

```diff
@@ -577,7 +577,7 @@ fn override_toolchain(cfg_override: &ConfigOverride) -> Result<RestoreToolchainC
             }
 
             let output_version = std::str::from_utf8(&output.stdout)?;
-            let version = parse_version(output_version);
+            let version = parse_version(output_version).unwrap();
             Ok(version)
```

Comment:
Given this has a potential to fail, it would be better to avoid panicking here:

```suggestion
            parse_version(output_version)
                .ok_or_else(|| anyhow!("Failed to parse the version of `{cmd_name}`"))
```